pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +191,78 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def encoder(in_channels: int = 3,) -> SequentialEncoder:
+    r"""Encoder part of the :class:`Transformer` from :cite:`SKL+2018`.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+
+    """
+    modules = (
+        nn.ReflectionPad2d(15),
+        ConvBlock(in_channels=in_channels, out_channels=32, kernel_size=3, stride=1),
+        ConvBlock(in_channels=32, out_channels=32, kernel_size=3, stride=2),
+        ConvBlock(in_channels=32, out_channels=64, kernel_size=3, stride=2),
+        ConvBlock(in_channels=64, out_channels=128, kernel_size=3, stride=2),
+        ConvBlock(in_channels=128, out_channels=256, kernel_size=3, stride=2),
+    )
+    return SequentialEncoder(modules)
+
+
+def decoder(
+    out_channels: int = 3, num_residual_blocks: int = 9,
+) -> pystiche.SequentialModule:
+
+    modules: List[nn.Module] = []
+    for i in range(num_residual_blocks):
+        modules.append(residual_block(256))
+
+    modules.append(UpsampleConvBlock(in_channels=256, out_channels=256, kernel_size=3))
+    modules.append(UpsampleConvBlock(in_channels=256, out_channels=128, kernel_size=3))
+    modules.append(UpsampleConvBlock(in_channels=128, out_channels=64, kernel_size=3))
+    modules.append(UpsampleConvBlock(in_channels=64, out_channels=32, kernel_size=3))
+    modules.append(nn.ReflectionPad2d(3))
+    modules.append(
+        conv(
+            in_channels=32,
+            out_channels=out_channels,
+            kernel_size=7,
+            stride=1,
+            padding="valid",
+        )
+    )
+    return pystiche.SequentialModule(*modules)
+
+
+class DecoderTanhOutput(nn.Module):
```

Comment:
I think its OK to do this as a local class in `Decoder` since it is useless otherwise.